stubs/tensorflow/tensorflow/keras/callbacks.pyi

```diff
@@ -0,0 +1,146 @@
+    def on_train_end(self, logs: Mapping[str, object] | None = None) -> None: ...
+
+class History(Callback):
+    history: dict[str, list[Any]]
+
+class ModelCheckpoint(Callback):
+    monitor_op: Any
+    filepath: str
+    _options: tf.train.CheckpointOptions | tf.saved_model.SaveOptions | None
+    def __init__(
+        self,
+        filepath: str,
+        monitor: str = "val_loss",
+        verbose: Literal[0, 1] = 0,
+        save_best_only: bool = False,
+        save_weights_only: bool = False,
+        mode: Literal["auto", "min", "max"] = "auto",
+        save_freq: str | int = "epoch",
+        options: tf.train.CheckpointOptions | tf.saved_model.SaveOptions | None = None,
+        initial_value_threshold: float | None = None,
+    ) -> None: ...
+    def _save_model(self, epoch: int, batch: int | None, logs: _Logs): ...
+
+class TensorBoard(Callback):
+    _start_batch: int
+    _stop_batch: int
+    _should_write_train_graph: bool
+    write_steps_per_second: bool
+    _batch_start_time: float
+    _train_step: tf.Variable
+    _should_trace: bool
+    _is_tracing: bool
+    _global_train_batch: int
+    update_freq: int | Literal["epoch"]
+
+    def __init__(
+        self,
+        log_dir: str = "logs",
+        histogram_freq: int = 0,
+        write_graph: bool = True,
+        write_images: bool = False,
+        write_steps_per_second: bool = False,
+        update_freq: int | Literal["epoch"] = "epoch",
+        profile_batch: int | tuple[int, int] = 0,
+        embeddings_freq: int = 0,
+        embeddings_metadata: dict[str, None] | None = None,
+    ) -> None: ...
+    def _write_keras_model_train_graph(self) -> None: ...
+    def _stop_trace(self, batch: int | None = None) -> None: ...
+
+class BackupAndRestore(Callback):
+    def __init__(
+        self, backup_dir: str, save_freq: str = "epoch", delete_checkpoint: bool = True, save_before_preemption: bool = False
+    ) -> None: ...
+
+class BaseLogger(Callback):
+    def __init__(self, stateful_metrics: Iterable[str] | None = None) -> None: ...
+
+class CSVLogger(Callback):
+    def __init__(self, filename: str, separator: str = ",", append: bool = False) -> None: ...
+
+class EarlyStopping(Callback):
+    monitor_op: Any
+    def __init__(
+        self,
+        monitor: str = "val_loss",
+        min_delta: float = 0,
+        patience: float = 0,
+        verbose: Literal[0, 1] = 0,
+        mode: Literal["auto", "min", "max"] = "auto",
+        baseline: float | None = None,
+        restore_best_weights: bool = False,
+        start_from_epoch: int = 0,
+    ) -> None: ...
+
+class LearningRateScheduler(Callback):
+    def __init__(self, schedule: LearningRateSchedule, verbose: Literal[0, 1] = 0) -> None: ...
+
+class ProgbarLogger(Callback):
+    use_steps: bool
+    def __init__(
+        self, count_mode: Literal["steps", "samples"] = "samples", stateful_metrics: Iterable[str] | None = None
+    ) -> None: ...
+
+class TerminateOnNaN(Callback): ...
```

Comment:
This file needs getattr incomplete (same comment for other files) as there are a few more callbacks still missing. Or just add the rest. There's only 4 missing callbacks.

Reply:
I've added the missing callbacks in c533ddc1c